pymc3/distributions/distribution.py

```diff
@@ -386,23 +387,37 @@ def random(self, point=None, size=None, **kwargs):
             not_broadcast_kwargs.update(**kwargs)
             if self.wrap_random_with_dist_shape:
                 size = to_tuple(size)
-                with _DrawValuesContextBlocker():
-                    test_draw = generate_samples(
-                        self.rand,
-                        size=None,
-                        not_broadcast_kwargs=not_broadcast_kwargs,
+                # We don't know anything about what self.rand returns so we
```

Comment:
We are capturing *all* exceptions here when trying to generate a test sample.  Should we:

- Log the exception? and/or
- Signal a warning about the exception?

Finally, should we simply give up at this point?  As I understand things, we are generating this single sample as a way to try to determine the shape of the RV in question. I *believe* that this is already a fall-back position.  If we have not been able to identify the shape here, should we give up and raise an exception, instead of trying to continue?  Couldn't this cause us to compute "garbage out" from "garbage in"? Or cause us to raise a cryptic exception *later* that the user will have a harder time diagnosing?